lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -80,6 +80,49 @@ static Type lowerType(Type type) {
   return {};
 }
 
+/// This verifies that the target operation has been lowered to a legal
+// operation.  This checks that the operation recursively has no FIRRTL
+// operations or types.
+static LogicalResult verifyOpLegality(Operation *op) {
+  SmallVector<Operation *> worklist = {op};
+  while (!worklist.empty()) {
```

Comment:
Would it be reasonable to just use `walk`?

It is generally ok to write things recursive on the region tree, so this doesn't need to be worklist driven. worklist is also fine tho

Reply:
I think walk is way better, I don't want to be adding all ops in some giant function to a worklist..